xarray/core/dataset.py

```diff
@@ -3558,6 +3557,13 @@ def drop(  # noqa: F811
                 ds = ds._drop_labels(labels, dim, errors=errors)
             return ds
         elif dim is None:
+            if isinstance(labels, DataArrayCoordinates):
+                warnings.warn(
+                    "dropping coordinates using a DataArrayCoordinates object "
+                    "is deprecated; use drop_vars or a list of coordinates",
+                    DeprecationWarning,
```

Comment:
`FutureWarning` is more appropriate here, since we will probably break existing code in the future by giving this different behavior:
```suggestion
                    FutureWarning,
```

Reply:
It also would be great to add a test that verifies that the warning is raised :)

Reply:
Should the second warning for `list`-like `labels` also be a `FutureWarning`?

Reply:
I think `DeprecationWarning` might be more appropriate there, since continuing to rely on that will result in errors rather than different behavior.